### PR TITLE
feat(ZARP Gaming): add presence

### PR DESCRIPTION
**Pull request by callumok2004**
## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

PUpOfIhU61 1

JIrIZuMI70 1

vuyURUhu96 1

MelAYeVu00 1

</details>

### CODE REVIEW

File: websites/Z/ZARP Gaming/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		"name": "Callum",
+		"id": "506899274748133376"
+	},
+	"service": "ZARP Gaming",
+	"description": {
+		"en": "ZARP Gaming is an online gaming community focused on garry's mod servers, such as DarkRP, TTT, Prophunt and more."
+	},
+	"url": [
+		"zarpgaming.com",
+		"www.zarpgaming.com",
```
Comment (ImRodry) [inline websites/Z/ZARP Gaming/dist/metadata.json:13]: this URL doesn't exist, remove it

File: websites/Z/ZARP Gaming/presence.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+				window.location.pathname == "/index.php" ||
+				window.location.pathname == "/index.php/forum"
+			)
+				presenceData.details = "Viewing the home page";
+
+			if (window.location.pathname.startsWith("/index.php/forum/")) {
+				presenceData.details = "Viewing a section:";
+				presenceData.state = (document.querySelector(
+					"#Kunena > form > div > div.kheader > h2 > span"
+				) as HTMLElement)?.innerText || document.title;
```
Comment (ImRodry) [inline websites/Z/ZARP Gaming/presence.ts:21]: don't use `as` and don't use HTMLElement, use the actual class name
```suggestion
				presenceData.state = document.querySelector<HTMLElement>(
					"#Kunena > form > div > div.kheader > h2 > span"
				)?.innerText ?? document.title;
```

File: websites/Z/ZARP Gaming/presence.ts [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+							.replace(/[-\s]+/g, " ");
+						break;
+					}
+				}
+			}
+
+			if (!presenceData.details) {
+				presenceData.details = "Viewing a page:";
+				presenceData.state = document.title;
+			}
```
Comment (ImRodry) [inline websites/Z/ZARP Gaming/presence.ts:132]: these should all be else ifs

File: websites/Z/ZARP Gaming/presence.ts [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+			if (
+				document.querySelector(
+					"#Kunena > div:nth-child(5) > div.kheader > h1 > span"
+				)
+			) {
+				presenceData.details = "Viewing a thead:";
+				presenceData.state = (document.querySelector(
+					"#Kunena > div:nth-child(5) > div.kheader > h1 > span"
+				) as HTMLElement).innerText.replace("TOPIC: ", "");
```
Comment (ImRodry) [inline websites/Z/ZARP Gaming/presence.ts:59]: HTMLElement is an abstract class, please use the proper type or none at all. Also don't use `as`

File: websites/Z/ZARP Gaming/presence.ts [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+				];
+			}
+
+			if (
+				window.location.pathname.startsWith("/index.php/2013-11-13-21-35-14/")
+			) {
+				presenceData.details = "Viewing a section:";
+				presenceData.state = (document.querySelector(
+					"#Kunena > div.kblock.kpathway.breadcrumbs-1 > div > div > div > div.path-element > a"
+				) as HTMLElement).innerText;
```
Comment (ImRodry) [inline websites/Z/ZARP Gaming/presence.ts:74]: ditto

File: websites/Z/ZARP Gaming/presence.ts [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+					"#Kunena > div.kblock.kpathway.breadcrumbs-1 > div > div > div > div:nth-child(3) > a"
+				) as HTMLElement).innerText;
+			}
+
+			if (window.location.pathname.includes("/new-topic")) {
+				presenceData.details = "Creating a topic in section:";
+				const topics = document.querySelector(
+					"#postcatid"
+				) as HTMLSelectElement;
+				for (let i = 0; i < topics.options.length; i++) {
```
Comment (ImRodry) [inline websites/Z/ZARP Gaming/presence.ts:119]:
```suggestion
				for (const option of topic.options) {
```

**Review by ImRodry — CHANGES_REQUESTED** [review]

**Comment by ImRodry** [comment]
Please don't resolve conversations without addressing them

File: websites/Z/ZARP Gaming/presence.ts [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+					break;
+				case "/index.php/help/faq":
+					presenceData.details = "Viewing the FAQ";
+					if (window.location.pathname.includes("faq/"))
+						presenceData.state = document.title;
+					break;
+				case "/index.php/profile/profile-edit":
+					presenceData.details = "Editing profile";
+					break;
+				case "/index.php/2013-11-13-21-35-14/":
```
Comment (ImRodry) [inline websites/Z/ZARP Gaming/presence.ts:35]: This doesn't seem right, why would you support a 2013 page and not others?

File: websites/Z/ZARP Gaming/presence.ts [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+						"#postcatid"
+					)) {
+						if (option.selected) {
+							presenceData.state = option.textContent
+								.replace(/^[-\s]+/, "")
+								.replace(/[-\s]+/g, " ");
+							break;
+						}
+					}
+					break;
```
Comment (ImRodry) [inline websites/Z/ZARP Gaming/presence.ts:60]: this should not be a for loop, you need to find the selected option and use it

File: websites/Z/ZARP Gaming/presence.ts [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+			}
+
+			if (!presenceData.details) {
+				presenceData.details = "Viewing a page:";
+				presenceData.state = document.title;
+			}
+			break;
+		case "control.zarpgaming.com":
+			presenceData.details = "Server Control Panel";
+			break;
```
Comment (ImRodry) [inline websites/Z/ZARP Gaming/presence.ts:115]: is that all? Can't you show more?

**Review by ImRodry — CHANGES_REQUESTED** [review]

File: websites/Z/ZARP Gaming/presence.ts (hunk 9 above)
Comment (callumok2004) [reply websites/Z/ZARP Gaming/presence.ts:115]: Nope, due to privacy content on this page cannot be shown regardless, though still showing you are on the control panel is nice to have for us.

File: websites/Z/ZARP Gaming/presence.ts (hunk 7 above)
Comment (callumok2004) [reply websites/Z/ZARP Gaming/presence.ts:35]: This is just show the forums is, it is correct and all tested.

Comment (ImRodry) [reply websites/Z/ZARP Gaming/presence.ts:35]: so everything on the forums is on this exact page with nothing after it? Show me a screenshot of that

File: websites/Z/ZARP Gaming/presence.ts (hunk 9 above)
Comment (ImRodry) [reply websites/Z/ZARP Gaming/presence.ts:115]: You can have a setting for that

File: websites/Z/ZARP Gaming/dist/metadata.json [hunk 10]
```diff
@@ -0,0 +1,28 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.7",
+	"author": {
+		"name": "Callum",
+		"id": "506899274748133376"
+	},
+	"service": "ZARP Gaming",
+	"description": {
+		"en": "ZARP Gaming is an online gaming community focused on garry's mod servers, such as DarkRP, TTT, Prophunt and more."
```
Comment (StrikerFRFX) [inline websites/Z/ZARP Gaming/dist/metadata.json:9]: Garry's Mod*

**Review by EncryptedDev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Comment by EncryptedDev** [comment]
Thank you for the PR

**Pull request merged**